### PR TITLE
Add npm-run-all package to simplify running scripts

### PR DESCRIPTION
**Pull request by mkaz**
## Description

The npm-run-all package allows for an easier way to run multiple scripts
from the package.json file. We are going to run a few scripts to
generate api docs, so instead of joining multiple commands using && it
can be simplified to creating a command for each and then running:

`npm-run-all api-docs:*`

Package docs:
https://www.npmjs.com/package/npm-run-all

## How has this been tested?

Run: `npm run docs:build` and confirm it runs the set of scripts specified.

## Types of changes

- Moves api doc generation scripts to own commands
- Move doc generation script to own command
- Uses npm-run-all in docs:build command to run others

### CODE REVIEW

**Comment by gziolo** [comment]
We can also update this line with the newly added command:

https://github.com/WordPress/gutenberg/blob/8f90816cfe7c19baba2f5fc9a54c61b2eb59b2e3/package.json#L309

Now that we generate the docs for blocks based on `block.json` files, we should also add an entry in `lint-staged` that watches for changes in those files and triggers a re-run of the documentation for that specific aspect.

https://github.com/WordPress/gutenberg/blob/8f90816cfe7c19baba2f5fc9a54c61b2eb59b2e3/package.json#L290

**Comment by mkaz** [comment]
Good catch, the npm-run-all works there also, so just added the `api-docs:*` wilcard so any future additions will also get picked up by the pre-commit hook. 👍

File: package.json [hunk 1]
```diff
@@ -306,7 +310,7 @@
 			"node ./docs/tool/index.js"
 		],
 		"packages/**/*.{js,ts,tsx}": [
```
Comment (gziolo) [inline package.json]: We need to include JSON files as well 😄 
```suggestion
		"packages/**/*.{js,ts,tsx,json}": [
```

Comment (mkaz) [reply package.json]: 😁

**Review by gziolo — APPROVED** [review]
💯

**Review by cbravobernal — APPROVED** [review]

**Pull request merged**